units/util.go

```diff
@@ -0,0 +1,45 @@
+package units
+
+import (
+	"context"
+
+	"github.com/evergreen-ci/evergreen"
+	"github.com/evergreen-ci/evergreen/model/host"
+	"github.com/mongodb/grip"
+	"github.com/pkg/errors"
+)
+
+func HandlePoisonedHost(ctx context.Context, env evergreen.Environment, h *host.Host, reason string) error {
+	if h == nil {
+		return errors.New("no host found")
+	}
+	if h.ParentID != "" {
+		parent, err := host.FindOneId(h.ParentID)
+		if err != nil {
+			return errors.Wrap(err, "error finding parent host")
+		}
+		if parent != nil {
+			containers, err := parent.GetActiveContainers()
+			if err != nil {
+				return errors.Wrap(err, "error getting containers")
+			}
+			catcher := grip.NewBasicCatcher()
+			for _, container := range containers {
+				catcher.Add(DisableAndNotifyPoisonedHost(ctx, env, container, reason))
+			}
+			catcher.Add(DisableAndNotifyPoisonedHost(ctx, env, *parent, reason))
+			if catcher.HasErrors() {
```

Comment:
If there aren't errors, that means that we've called `DisableAndNotifyPoisonedHost` on `h` already, since it's one of the `containers`. I think that means we should just `return catcher.Resolve()` without checking `if catcher.HasErrors()`.

Reply:
I initially misread this as needing to handle a race, if a different container was also poisoned (likely if the disk is full), and we ended up running this twice for the same parent host. I think the solution to that also handles what you're saying, which is to continue on error in this function and return all errors to the caller